algo/src/main/java/org/neo4j/graphalgo/impl/MSClosenessCentrality.java

```diff
@@ -0,0 +1,88 @@
+package org.neo4j.graphalgo.impl;
+
+import org.neo4j.graphalgo.api.Graph;
+import org.neo4j.graphalgo.impl.msbfs.BfsConsumer;
+import org.neo4j.graphalgo.impl.msbfs.MultiSourceBFS;
+import org.neo4j.graphdb.Direction;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicIntegerArray;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+/**
+ *
+ * Normalized Closeness Centrality
+ *
+ * @author mknblch
+ */
+public class MSClosenessCentrality {
+
+    private final Graph graph;
+
+    private final ExecutorService executorService;
+
+    private final AtomicIntegerArray farness;
+
+    private final int nodeCount;
+
+    public MSClosenessCentrality(Graph graph, ExecutorService executorService) {
+        this.graph = graph;
+        nodeCount = graph.nodeCount();
+        this.executorService = executorService;
+        farness = new AtomicIntegerArray(nodeCount);
+    }
+
+    public MSClosenessCentrality compute() {
+
+        final BfsConsumer consumer = (nodeId, depth, sourceNodeIds) -> {
+            // TODO size of sourceNodeIds
+            while (sourceNodeIds.hasNext()) {
+                sourceNodeIds.next();
+                farness.addAndGet(nodeId, depth);
```

Comment:
if so, we can add or modify the third parameter to return the size w/o having to iterate.